### DISCUSSION TITLE
[✨] Expose next output in middleware for ISR (Incremental Static Regeneration)

### DISCUSSION THREAD

**Discussion by fprl** [QwikDev/qwik-evolution#129] [Proposals For Qwik]
### Is your feature request related to a problem?

Not a problem but a performance improvement using node/deno adapter by using ISR (Incremental Static Regeneration).

### Describe the solution you'd like

It would be nice if Qwik could expose the `next` output in the middleware.

Right now is of type `next: () => Promise<void>` where it could be `next: MiddlewareNext<Response | EndpointOutput>`. With this, we could cache the response and take advantage of ISR.

You can see an [example with Astro here](https://deploy-preview-3527--astro-docs-2.netlify.app/en/guides/incremental-static-regeneration/#overview).

PS: It would be nice to also have a section like this in the docs, explaining how to achieve ISR with different adapters.

Thank you.

### Describe alternatives you've considered

Maybe an alternative would be to expose some kind of method in the middleware request event.

**Comment by gioboa** [comment]
Hi @fprl would you like to create a PR for that?

**Comment by fprl** [comment]
Hi @gioboa, sure. I'm trying to understand how things works

**Comment by gioboa** [comment]
If you have any questions you can ping me on Qwik discord. I'm more than happy to help you

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by adaliszk** [comment]
I am also looking for something like this, but not specifically the cache middleware direction but simply limiting the actual SSG site generation to the ones that did changed. So far I had been able to limit the dynamic portion of a Quick City application and merge the new pages into the existing ones provided that all referencing pages got regenerated as well.

Its a bit of a work, but I can see something smart being implemented for v2 or with a generic content provider solution.
